### PR TITLE
refactor clusterrole for kube-ovn

### LINKED ISSUE #2723: Remove Unnecessary Permissions from ClusterRole
Some permission is not used by Kube-OVN we should remove them

### PR DESCRIPTION
**Pull request by hongzhen-ma**
- [ ] Make sure you have followed [Kube-OVN Code Style](https://github.com/kubeovn/kube-ovn/blob/master/CODE_STYLE.md).

### What type of this PR
- security

原 sa 拆分为四个，分别为 ovn/ovn-ovs/kube-ovn-cni/kube-ovn-app
其中 ovn，考虑兼容性，为保留的原 sa 的名称，kube-onv-controller 绑定。同时，speaker、webhook 这种可以保证正常启动。
ovn-ovs， 为 ovn-central 和 ovs-ovn pod 启动，绑定的 serviceAccount
kube-ovn-cni，为 kube-ovn-cni 绑定使用
kube-ovn-app，为 kube-ovn-pinger 和 kube-ovn-monitor 绑定使用

### Which issue(s) this PR fixes:
Fixes #2723 
### <samp>🤖 Generated by Copilot at a546417</samp>

Improved the reliability of the e2e test for ovn-vpc-nat-gw by adding a readiness check for ovn eip objects.

### <samp>🤖 Generated by Copilot at a546417</samp>

> _Create `ovn eip`_
> _Wait for readiness check - poll_
> _Autumn tests are green_

### <samp>🤖 Generated by Copilot at a546417</samp>

* Replace `ovnEipClient.CreateSync` with `ovnEipClient.Create` and a `wait.PollImmediate` block to check the `Ready` status of the ovn eip object after creating it ([link](https://github.com/kubeovn/kube-ovn/pull/2833/files?diff=unified&w=0#diff-c1831d6aa37f96132386404bff7ef6c738c40bcef66ba7492aca28929d61dca2L393-R405))
* Import the `wait` package to use the `wait.PollImmediate` function ([link](https://github.com/kubeovn/kube-ovn/pull/2833/files?diff=unified&w=0#diff-c1831d6aa37f96132386404bff7ef6c738c40bcef66ba7492aca28929d61dca2R17))

### CODE REVIEW

File: charts/templates/ovn-CR.yaml [hunk 1]
```diff
@@ -62,19 +63,35 @@ rules:
...
       - networking.k8s.io
       - apps
       - extensions
     resources:
       - networkpolicies
-      - services
       - services/status
+      - daemonsets
+    verbs:
+      - update
```
Comment (oilbeater) [inline charts/templates/ovn-CR.yaml:82]: update 权限应该不需要

File: charts/templates/ovn-CR.yaml [hunk 2]
```diff
@@ -62,19 +63,35 @@ rules:
...
+      - ""
+      - networking.k8s.io
+      - apps
+      - extensions
+    resources:
+      - services
       - endpoints
       - statefulsets
-      - daemonsets
       - deployments
```
Comment (oilbeater) [inline charts/templates/ovn-CR.yaml:95]: create/delete/update/patch 应该都不需要

**Review by oilbeater — CHANGES_REQUESTED** [review]

File: charts/templates/ovn-CR.yaml (hunk 1 above)
Comment (hongzhen-ma) [reply charts/templates/ovn-CR.yaml]: services/status 需要update 权限，已单独抽出

File: charts/templates/ovn-CR.yaml (hunk 2 above)
Comment (hongzhen-ma) [reply charts/templates/ovn-CR.yaml:99]: endpoints 更新中，需要对 services 做 update 操作
vpc-nat-gw pod 是 sts 部署，需要 statefulsets 的权限
load-balancer service，创建deployment pod，需要deployment 相关权限
vpc switch-lb-rule 创建svc，并且创建endpoints，需要 endpoints 相关权限

基于以上考虑，这些参数放在一起，没有再细分

**Review by oilbeater — APPROVED** [review]

**Pull request merged**